.github/workflows/spellcheck.yaml

```diff
@@ -0,0 +1,58 @@
+---
+name: spell and grammer check
+on:
+  pull_request:
+    paths:
+      - "docs"         #(provide name of the file or folder)
+jobs:
+  check-spelling:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    # Install the dependencies for grammer check and spell check.
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip setuptools
+        python -m pip install pyspelling 
+        sudo apt-get install hunspell hunspell-en-us aspell aspell-en
+        pip install pylanguagetool
+      
+    - uses: actions/checkout@v2
+    - name: Spellcheck
+      uses: rojopolis/spellcheck-github-actions@0.34.0
+      with:
+        config_path: .github/.spellcheck.yaml
+        source_files: "docs"       # name of the file to check spell and grammer)
+        task_name: Markdown
+    # To check the grammatical mistakes of the given files and folders.
+    - name: Run grammar check 
+      run: |
+        # Read the text from  all files in the  directory and its subdirectories
+        files_to_check=$(find docs -type f)   
+
+        # Define the LanguageTool API URL and language (e.g., en-US for English)
+        api_url="https://languagetool.org/api/v2/check"
```

Comment:
We can replace this with npm-modules/python-packages providing the same functionality in the future